### PR TITLE
Duplicate storage

### PR DESCRIPTION
**Pull request by VijayP30**
Resolves:
- #11 
- #24 
- #25 

I created a set object to keep track of the horoscope objects if you stringify them. If the stringified object is not in the set already, we should add the horoscope object to the list to be displayed, if it is in the set, then we do not add the duplicate. I wrote additional unit tests for coverage of this new feature.

### CODE REVIEW

File: source/horoscope/js/horoscope.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ async function init() {
         let category = categoryElement.value;
         let message = fortuneElement.innerText;
         let today = new Date();
+        today.setHours(0,0,0,0);
```
Comment (Miyuki-L) [inline source/horoscope/js/horoscope.js:51]: Why do we need this?  This sets the hours in the date object to 00:00:00

Causes the "min ago" of the past history to be wrong

File: source/horoscope/js/sidebar.js [hunk 2]
```diff
@@ -39,6 +39,7 @@ class Horoscope{
 }
 
 let horoscopes = [];
+let horoscopesJSON = new Set();
```
Comment (Miyuki-L) [inline source/horoscope/js/sidebar.js:42]: Do we need a whole separate variable?

**Review by Miyuki-L — CHANGES_REQUESTED** [review]
One of the changes break the timing for the past history. 

Also would it not be an option (or an additional safety meausure) to just stopping the user from clicking the save button after saving and reenable it after new reading.

File: source/horoscope/js/horoscope.js (hunk 1 above)
Comment (VijayP30) [reply source/horoscope/js/horoscope.js]: i removed this line. i am excluding the date field when checking if the horoscope exists in the saveHoroscope function itself now. The time function looks like it is working correctly now.

File: source/horoscope/js/sidebar.js (hunk 2 above)
Comment (VijayP30) [reply source/horoscope/js/sidebar.js:42]: We can either have a separate variable, or we can loop through the horoscopes array every time we save a horoscope. Tradeoff for time vs memory. If there is a better way to compare objects in JS, let me know.

File: source/horoscope/js/sidebar.js
Comment (StephenTan12) [inline source/horoscope/js/sidebar.js:1]: gotta clear the horoscopeJson to ensure consistency

**Review by StephenTan12 — CHANGES_REQUESTED** [review]
Just have one minor issue. Other than that, looks good

File: source/horoscope/js/sidebar.js
Comment (VijayP30) [reply source/horoscope/js/sidebar.js:1]: done, good catch. thanks

**Review by StephenTan12 — APPROVED** [review]
👍

**Pull request merged**